nodes/src/nodes/tool_git/IGlobal.py

```
@@ -0,0 +1,130 @@
+            ssh_passphrase=ssh_passphrase,
+            safe_mode=safe_mode,
+        )
+
+        if _is_url(repo_path):
+            # Auto-clone the remote URL into a fresh temp directory
+            tmp = tempfile.mkdtemp(prefix='rocketride_git_')
+            try:
+                git.clone(url=repo_path, path=tmp)
+            except (GitError, Exception) as exc:
+                shutil.rmtree(tmp, ignore_errors=True)
+                raise ValueError(f'tool_git: failed to clone {repo_path!r}: {exc}') from exc
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)